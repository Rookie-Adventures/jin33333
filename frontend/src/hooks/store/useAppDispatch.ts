import { useDispatch } from 'react-redux';

import type { AppDispatch } from '@/store';

/**
 * 类型安全的dispatch hook
 * @returns {AppDispatch} 类型安全的dispatch函数
 */
export const useAppDispatch = () => useDispatch<AppDispatch>();
