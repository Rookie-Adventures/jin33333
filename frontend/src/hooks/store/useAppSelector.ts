import { useSelector } from 'react-redux';
import type { TypedUseSelectorHook } from 'react-redux';

import type { RootState } from '@/store';

/**
 * 类型安全的selector hook
 * @type {TypedUseSelectorHook<RootState>} 类型安全的selector函数
 */
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector; 