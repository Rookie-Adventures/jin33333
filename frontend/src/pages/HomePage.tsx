import { Box } from '@mui/material';
import { type FC } from 'react';

import FeatureSection from '../components/business/sections/FeatureSection';
import HeroSection from '../components/business/sections/HeroSection';
import PricingSection from '../components/business/sections/PricingSection';
import Footer from '../components/common/layout/Footer';
import Navbar from '../components/common/layout/Navbar';

const HomePage: FC = () => {
  return (
    <Box
      sx={{
        width: '100vw',
        margin: '-8px',
        padding: 0,
        overflow: 'hidden',
        position: 'relative',
        left: '50%',
        right: '50%',
        marginLeft: '-50vw',
        marginRight: '-50vw',
      }}
    >
      <Navbar />
      {/* Hero Section */}
      <Box
        sx={{
          background: 'linear-gradient(135deg, #24243e 0%, #302b63 50%, #0f0c29 100%)',
          color: 'white',
          position: 'relative',
          pt: '64px', // 为导航栏留出空间
          '&::before': {
            content: '""',
            position: 'absolute',
            top: 0,
            left: 0,
            right: 0,
            bottom: 0,
            background:
              'radial-gradient(circle at 50% 50%, rgba(255,255,255,0.1) 0%, rgba(0,0,0,0) 70%)',
            pointerEvents: 'none',
          },
        }}
      >
        <HeroSection />
      </Box>

      {/* Feature Section */}
      <Box>
        <FeatureSection />
      </Box>

      {/* Pricing Section */}
      <Box
        sx={{
          background: 'linear-gradient(135deg, #f5f7fa 0%, #e8ecf3 100%)',
        }}
      >
        <PricingSection />
      </Box>
      <Footer />
    </Box>
  );
};

export default HomePage;
