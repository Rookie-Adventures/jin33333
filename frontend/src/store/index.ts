import { configureStore } from '@reduxjs/toolkit';
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import appReducer from './slices/appSlice';
import authReducer from './slices/authSlice';

// 配置持久化
const persistConfig = {
  key: 'root',
  version: 1,
  storage,
  whitelist: ['app', 'auth'], // 只持久化这些 reducer
};

const persistedReducer = persistReducer(persistConfig, appReducer);

export const store = configureStore({
  reducer: {
    app: persistedReducer,
    auth: authReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
});

export const persistor = persistStore(store);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector; 