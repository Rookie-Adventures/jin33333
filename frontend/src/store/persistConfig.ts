import type { Transform, PersistConfig, PersistedState } from 'redux-persist';
import storage from 'redux-persist/lib/storage';

// 定义可序列化的状态类型
type SerializableValue = string | number | boolean | null | undefined | SerializableObject | Date;
interface SerializableObject {
  [key: string]: SerializableValue;
}

// 定义转换器配置类型
interface TransformConfig {
  debug?: boolean;
  whitelist?: string[];
}

// 转换单个值
function transformSingleValue(value: unknown): unknown {
  if (value instanceof Date) {
    return value.toISOString();
  }
  if (typeof value === 'string' && /^\d{4}-\d{2}-\d{2}/.test(value)) {
    return new Date(value);
  }
  return value;
}

// 转换对象值
function transformObjectValue(
  value: Record<string, unknown>
): Record<string, unknown> {
  return Object.entries(value).reduce(
    (acc, [k, v]) => ({
      ...acc,
      [k]: v && typeof v === 'object' && !(v instanceof Date)
        ? transformObjectValue(v as Record<string, unknown>)
        : transformSingleValue(v)
    }),
    {}
  );
}

// 转换状态
function transformState<T extends Record<string, unknown>>(
  state: T
): T {
  if (!state) return state;
  return transformObjectValue(state) as T;
}

// 创建日期转换器
function createDateTransform<T extends Record<string, unknown>>(): Transform<T, T, T, T> {
  return {
    in: (state: T, _key: keyof T, _config: TransformConfig): T => {
      return transformState(state);
    },
    out: (state: T, _key: keyof T, _config: TransformConfig): T => {
      return transformState(state);
    }
  };
}

// 创建基础配置
export function createPersistConfig<T extends Record<string, unknown>>(key: string): PersistConfig<T> {
  const debug = process.env.NODE_ENV !== 'production';
  const defaultPersistState = {
    _persist: { version: 1, rehydrated: true }
  } as T & PersistedState;
  
  return {
    key,
    storage,
    whitelist: ['auth', 'app'],
    transforms: [createDateTransform<T>()],
    version: 1,
    migrate: (state: PersistedState): Promise<T & PersistedState> => {
      if (!state) {
        return Promise.resolve(defaultPersistState);
      }
      return Promise.resolve({
        ...state,
        ...defaultPersistState._persist
      } as T & PersistedState);
    },
    debug,
  };
} 