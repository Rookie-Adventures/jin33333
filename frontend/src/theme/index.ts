import { createTheme as createMuiTheme } from '@mui/material/styles';
import type { ThemeOptions } from '@mui/material/styles';

export const defaultTheme: ThemeOptions = {
  palette: {
    mode: 'light' as const,
    primary: {
      main: '#1976d2',
      light: '#42a5f5',
      dark: '#1565c0',
      contrastText: '#ffffff'
    },
    secondary: {
      main: '#9c27b0',
      light: '#ba68c8',
      dark: '#7b1fa2',
      contrastText: '#ffffff'
    },
    error: {
      main: '#d32f2f',
      light: '#ef5350',
      dark: '#c62828',
      contrastText: '#ffffff'
    },
    warning: {
      main: '#ed6c02',
      light: '#ff9800',
      dark: '#e65100',
      contrastText: '#ffffff'
    },
    info: {
      main: '#0288d1',
      light: '#03a9f4',
      dark: '#01579b',
      contrastText: '#ffffff'
    },
    success: {
      main: '#2e7d32',
      light: '#4caf50',
      dark: '#1b5e20',
      contrastText: '#ffffff'
    },
    background: {
      default: '#ffffff',
      paper: '#f5f5f5'
    },
    text: {
      primary: 'rgba(0, 0, 0, 0.87)',
      secondary: 'rgba(0, 0, 0, 0.6)'
    }
  },
  typography: {
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    h1: {
      fontSize: '2.5rem',
      fontWeight: 500
    },
    h2: {
      fontSize: '2rem',
      fontWeight: 500
    },
    h3: {
      fontSize: '1.75rem',
      fontWeight: 500
    },
    h4: {
      fontSize: '1.5rem',
      fontWeight: 500
    },
    h5: {
      fontSize: '1.25rem',
      fontWeight: 500
    },
    h6: {
      fontSize: '1rem',
      fontWeight: 500
    },
    body1: {
      fontSize: '1rem',
      lineHeight: 1.5
    },
    body2: {
      fontSize: '0.875rem',
      lineHeight: 1.43
    },
    button: {
      fontSize: '0.875rem',
      textTransform: 'uppercase',
      fontWeight: 500
    }
  },
  spacing: 8,
  breakpoints: {
    values: {
      xs: 0,
      sm: 600,
      md: 960,
      lg: 1280,
      xl: 1920
    }
  },
  transitions: {
    easing: {
      easeInOut: 'cubic-bezier(0.4, 0, 0.2, 1)',
      easeOut: 'cubic-bezier(0.0, 0, 0.2, 1)',
      easeIn: 'cubic-bezier(0.4, 0, 1, 1)',
      sharp: 'cubic-bezier(0.4, 0, 0.6, 1)'
    },
    duration: {
      shortest: 150,
      shorter: 200,
      short: 250,
      standard: 300,
      complex: 375,
      enteringScreen: 225,
      leavingScreen: 195
    }
  },
  components: {
    MuiButton: {
      styleOverrides: {
        root: {
          textTransform: 'none'
        } as const
      }
    },
    MuiTextField: {
      defaultProps: {
        variant: 'outlined'
      }
    },
    MuiCard: {
      styleOverrides: {
        root: {
          borderRadius: 8
        }
      }
    }
  }
};

export const darkTheme: ThemeOptions = {
  ...defaultTheme,
  palette: {
    ...defaultTheme.palette,
    mode: 'dark' as const,
    background: {
      default: '#121212',
      paper: '#1e1e1e'
    },
    text: {
      primary: '#ffffff',
      secondary: 'rgba(255, 255, 255, 0.7)'
    },
    action: {
      active: '#fff',
      hover: 'rgba(255, 255, 255, 0.08)',
      selected: 'rgba(255, 255, 255, 0.16)',
      disabled: 'rgba(255, 255, 255, 0.3)',
      disabledBackground: 'rgba(255, 255, 255, 0.12)'
    }
  }
};

export const createTheme = (isDark: boolean, customOptions: Partial<ThemeOptions> = {}): ThemeOptions => {
  const baseTheme = isDark ? darkTheme : defaultTheme;
  return createMuiTheme(baseTheme, customOptions);
};

// 导出默认主题实例
export const theme = createTheme(false); 