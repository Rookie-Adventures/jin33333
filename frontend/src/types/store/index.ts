import { User } from '../auth';

// 根状态类型
export interface RootState {
  auth: AuthState;
  ui: UIState;
  app: AppState;
}

// 认证状态
export interface AuthState {
  user: User | null;
  token: string | null;
  loading: boolean;
  error: string | null;
}

// UI状态
export interface UIState {
  theme: 'light' | 'dark';
  sidebarCollapsed: boolean;
  modalStack: string[];
  loading: {
    [key: string]: boolean;
  };
}

// 应用状态
export interface AppState {
  initialized: boolean;
  version: string;
  environment: string;
  settings: {
    language: string;
    timezone: string;
    notifications: boolean;
  };
}

// Action类型
export interface Action<T = unknown> {
  type: string;
  payload?: T;
  error?: boolean;
  meta?: Record<string, unknown>;
}

// Thunk Action类型
export type ThunkAction<R = void, S = RootState> = (
  dispatch: Dispatch,
  getState: () => S
) => R;

// Dispatch类型
export type Dispatch = <T = unknown>(action: Action<T> | ThunkAction) => Promise<T> | T; 