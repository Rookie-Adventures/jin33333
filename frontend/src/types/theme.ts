export type ThemeMode = 'light' | 'dark';

export interface ThemeCustomizations {
  primaryColor: string;
}

export interface ThemeState {
  mode: ThemeMode;
  customizations: ThemeCustomizations;
} 