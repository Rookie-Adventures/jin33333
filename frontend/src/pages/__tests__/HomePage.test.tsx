import React from 'react';
import { render, screen } from '@testing-library/react';
import { ThemeProvider } from '@mui/material/styles';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { vi } from 'vitest';
import mediaQuery from 'css-mediaquery';

import HomePage from '../HomePage';
import { theme } from '@/theme';
import { store } from '@/store';

// Mock 子组件
vi.mock('@/components/common/layout/Navbar', () => ({
  default: () => <div data-testid="mock-navbar">Navbar</div>
}));

vi.mock('@/components/business/sections/HeroSection', () => ({
  default: () => <div data-testid="mock-hero-section">HeroSection</div>
}));

vi.mock('@/components/business/sections/FeatureSection', () => ({
  default: () => <div data-testid="mock-feature-section">FeatureSection</div>
}));

vi.mock('@/components/business/sections/PricingSection', () => ({
  default: () => <div data-testid="mock-pricing-section">PricingSection</div>
}));

// 创建 matchMedia mock
function createMatchMedia(width: number) {
  return (query: string) => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: vi.fn(),
    removeListener: vi.fn(),
    addEventListener: vi.fn(),
    removeEventListener: vi.fn(),
    dispatchEvent: vi.fn(),
  });
}

describe('HomePage', () => {
  const renderHomePage = (isDarkMode = false) => {
    return render(
      <Provider store={store}>
        <BrowserRouter>
          <ThemeProvider theme={theme(isDarkMode)}>
            <HomePage />
          </ThemeProvider>
        </BrowserRouter>
      </Provider>
    );
  };

  beforeEach(() => {
    // 重置 window.matchMedia
    window.matchMedia = createMatchMedia(1024);
  });

  it('应该正确渲染所有主要部分', () => {
    renderHomePage();

    // 验证所有主要部分都被渲染
    expect(screen.getByTestId('mock-navbar')).toBeInTheDocument();
    expect(screen.getByTestId('mock-hero-section')).toBeInTheDocument();
    expect(screen.getByTestId('mock-feature-section')).toBeInTheDocument();
    expect(screen.getByTestId('mock-pricing-section')).toBeInTheDocument();
  });

  it('应该正确应用布局样式', () => {
    renderHomePage();

    // 获取主容器
    const container = screen.getByTestId('mock-hero-section').parentElement;
    expect(container).toHaveStyle({
      background: 'linear-gradient(135deg, #24243e 0%, #302b63 50%, #0f0c29 100%)',
      color: 'white',
      position: 'relative'
    });
  });

  it('应该正确设置容器宽度和溢出处理', () => {
    const { container } = renderHomePage();
    
    const mainContainer = container.firstChild as HTMLElement;
    expect(mainContainer).toHaveStyle({
      width: '100vw',
      overflow: 'hidden'
    });
  });

  it('应该在暗色主题下正确渲染', () => {
    renderHomePage(true);
    
    const mainContainer = screen.getByTestId('mock-hero-section').parentElement;
    expect(mainContainer).toHaveStyle({
      background: 'linear-gradient(135deg, #24243e 0%, #302b63 50%, #0f0c29 100%)'
    });
  });

  it('应该在移动端视图下正确渲染', () => {
    // 模拟移动端视口
    window.matchMedia = createMatchMedia(375);
    
    renderHomePage();
    
    const mainContainer = screen.getByTestId('mock-hero-section').parentElement;
    expect(mainContainer).toHaveStyle({
      paddingTop: '64px' // 导航栏空间
    });
  });

  // 添加快照测试
  it('应该匹配快照', () => {
    const { container } = renderHomePage();
    expect(container).toMatchSnapshot();
  });

  it('应该在暗色主题下匹配快照', () => {
    const { container } = renderHomePage(true);
    expect(container).toMatchSnapshot();
  });

  it('应该在移动端视图下匹配快照', () => {
    window.matchMedia = createMatchMedia(375);
    const { container } = renderHomePage();
    expect(container).toMatchSnapshot();
  });
}); 