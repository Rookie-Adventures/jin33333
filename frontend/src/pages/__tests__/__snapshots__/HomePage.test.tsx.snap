// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage > 应该匹配快照 1`] = `
<div>
  <div
    class="MuiBox-root css-7tuhkw"
  >
    <div
      data-testid="mock-navbar"
    >
      Navbar
    </div>
    <div
      class="MuiBox-root css-se6w4j"
    >
      <div
        data-testid="mock-hero-section"
      >
        HeroSection
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        data-testid="mock-feature-section"
      >
        FeatureSection
      </div>
    </div>
    <div
      class="MuiBox-root css-1qim88"
    >
      <div
        data-testid="mock-pricing-section"
      >
        PricingSection
      </div>
    </div>
  </div>
</div>
`;

exports[`HomePage > 应该在暗色主题下匹配快照 1`] = `
<div>
  <div
    class="MuiBox-root css-7tuhkw"
  >
    <div
      data-testid="mock-navbar"
    >
      Navbar
    </div>
    <div
      class="MuiBox-root css-se6w4j"
    >
      <div
        data-testid="mock-hero-section"
      >
        HeroSection
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        data-testid="mock-feature-section"
      >
        FeatureSection
      </div>
    </div>
    <div
      class="MuiBox-root css-1qim88"
    >
      <div
        data-testid="mock-pricing-section"
      >
        PricingSection
      </div>
    </div>
  </div>
</div>
`;

exports[`HomePage > 应该在移动端视图下匹配快照 1`] = `
<div>
  <div
    class="MuiBox-root css-7tuhkw"
  >
    <div
      data-testid="mock-navbar"
    >
      Navbar
    </div>
    <div
      class="MuiBox-root css-se6w4j"
    >
      <div
        data-testid="mock-hero-section"
      >
        HeroSection
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        data-testid="mock-feature-section"
      >
        FeatureSection
      </div>
    </div>
    <div
      class="MuiBox-root css-1qim88"
    >
      <div
        data-testid="mock-pricing-section"
      >
        PricingSection
      </div>
    </div>
  </div>
</div>
`;
